googlesource/source_intergration_test.go

```diff
@@ -121,16 +128,107 @@ func cleanupDataSet() (err error) {
 	}
 	defer client.Close()
 
-	table := client.Dataset(datasetID).Table(tableID)
-	if err := table.Delete(ctx); err != nil {
+	meta := &bigquery.DatasetMetadata{
+		Location: location, // See https://cloud.google.com/bigquery/docs/locations
+	}
+
+	// create dataset
+	if err := client.Dataset(datasetID).Create(ctx, meta); err != nil && !strings.Contains(err.Error(), "duplicate") {
 		return err
 	}
+	fmt.Println("Dataset created")
+
+	sampleSchema := bigquery.Schema{
+		{Name: "name", Type: bigquery.StringFieldType},
+		{Name: "age", Type: bigquery.IntegerFieldType},
+		{Name: "updatedat", Type: bigquery.TimestampFieldType},
+	}
 
-	table = client.Dataset(datasetID).Table(tableID2)
-	if err := table.Delete(ctx); err != nil {
+	metaData := &bigquery.TableMetadata{
+		Schema:         sampleSchema,
+		ExpirationTime: time.Now().AddDate(1, 0, 0), // Table will be automatically deleted in 1 year.
+	}
+	tableRef := client.Dataset(datasetID).Table(tableIDTimeStamp)
+	err = tableRef.Create(ctx, metaData)
+	fmt.Println("Error: ", err)
+	if err != nil && !strings.Contains(err.Error(), "duplicate") {
 		return err
 	}
 
+	var query string
+
+	for i := 0; i < 8; i++ {
+		iString := fmt.Sprintf("%d", i)
+		query = "INSERT INTO `" + projectID + "." + datasetID + "." + tableIDTimeStamp + "`  values ('name" + iString +
+			"', " + iString + ", '" + time.Now().UTC().AddDate(0, 0, -i).Format("2006-01-02 15:04:05.999999 MST") + "' )"
+
+		q := client.Query(query)
+		q.Location = location
+
+		job, err := q.Run(ctx)
+		if err != nil {
+			log.Println("Error found: ", err)
+		}
+
+		status, err := job.Wait(ctx)
+		if err != nil {
+			log.Println("Error found: ", err)
+		}
+
+		if err := status.Err(); err != nil {
+			log.Println("Error found: ", err)
+		}
+	}
+
+	return nil
+}
+
+func dataUpdationWithTimestamp() {
+	ctx := context.Background()
+
+	client, err := bigquery.NewClient(ctx, projectID, option.WithCredentialsFile(serviceAccount))
+	if err != nil {
+		log.Println("Error found: ", err)
```

Comment:
Should this fail the test?

Reply:
It's good that we are now using t.log here, but we still have the question: if we cannot create the client to set up the test data, shouldn't we fail the test?

Also, there's a typo in the file name: inteRgration.

Reply:
data setup is a prerequisite for test cases to run. Should an error in doing that conclude that the test case failed? Because it's the issue with data setup and not the actual code.
@hariso what are your thoughts on this?

Reply:
Make sense. In that case, I will fail the test case if the data setup fails.

Reply:
implemented it in a similar fashion https://github.com/neha-Gupta1/conduit-connector-bigquery/blob/010f52e97ba2eb1487f3acd656632abfb98b6ca7/googlesource/source_intergration_test.go#L247